docs/RFCS/20200513_long_running_migrations.md

```diff
@@ -0,0 +1,473 @@
+- Feature Name: Long Running Migrations
+- Status: in-progress
+- Start Date: 2020-05-13
+- Authors: Irfan Sharif, Tobias Grieger, Dan Harrison
+- RFC PR: (PR # after acceptance of initial draft)
+- Cockroach Issue: [#39182](https://github.com/cockroachdb/cockroach/issues/39182)
+
+# Motivation
+
+CRDB gets older by the day, with each release bringing in changes that augment,
+remove, or replace functionality that are incompatible with the previous
+release. The only primitive available today to avoid undesirable behavior while
+clusters are running in mixed version modes is the cluster version setting (we
+provide a recap of how it works below). It lets us delay activating new
+functionality (that only the new binary version understands) until the operator
+guarantees to us that there are no nodes in the cluster running the old binary
+(and going forward, there never will be), at which point we're free to activate
+the features added in the new version.
+
+However, it's usually difficult to determine when legacy code can be deleted.
+This is because the primitive only "flips the switch" to allow new
+functionality; there's no infrastructure for moving out of the old world. As a
+result, legacy code piles up, and overlapping changes to the codebase are held
+up or even blocked. Consider the following example: TODO.
```

Comment:
Hehe, you should use the system.namespace migration, which is causing all sorts of horrible problems due to people including me misunderstanding how our current migrations system works. https://github.com/cockroachdb/cockroach/issues/49092